miniwin/sdl3gpu/src/miniwin_d3drmviewport.cpp

```diff
@@ -42,11 +42,57 @@ static void D3DRMMatrixMultiply(D3DRMMATRIX4D out, const D3DRMMATRIX4D a, const
 	}
 }
 
+void D3DRMMatrixInvert(D3DRMMATRIX4D out, const D3DRMMATRIX4D m)
+{
+	for (int i = 0; i < 3; ++i) {
+		for (int j = 0; j < 3; ++j) {
+			out[i][j] = m[j][i];
+		}
+	}
+
+	out[0][3] = out[1][3] = out[2][3] = 0.f;
+	out[3][3] = 1.f;
+
+	D3DVECTOR t = {m[3][0], m[3][1], m[3][2]};
+
+	out[3][0] = -(out[0][0] * t.x + out[1][0] * t.y + out[2][0] * t.z);
+	out[3][1] = -(out[0][1] * t.x + out[1][1] * t.y + out[2][1] * t.z);
+	out[3][2] = -(out[0][2] * t.x + out[1][2] * t.y + out[2][2] * t.z);
```

Comment:
This only inverts orthogonal matrices.
Perhaps rename the function to make this clear?

(An easy way to invert any matrix is [using cofactors](https://www.mathsisfun.com/algebra/matrix-inverse-minors-cofactors-adjugate.html).)

Reply:
I think I need to learn the basics of matrices before anything about them becomes easy.